buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTest.kt

```diff
@@ -54,7 +54,12 @@ abstract class DistributionTest : Test() {
 
     @get:Internal
     @get:Option(option = "rerun", description = "Always rerun the task")
-    val rerun: Property<Boolean> = project.objects.property(Boolean::class.javaObjectType).convention(false)
+    val rerun: Property<Boolean> = project.objects.property(Boolean::class.javaObjectType).convention(project.providers.systemProperty("idea.active").map { true }.orElse(false) )
```

Comment:
Could be simplified using
```kotlin
project.objects.property<Boolean>()....
```

Also, `.orElse(false)` will eagerly observe the system property at configuration time defeating instant execution cache.
It should be `.orElse(project.providers.provider { false })`.

Pretty unconvenient, we'd need some convenience like `provider.asPresence(): Provider<Boolean>` to replace `.map {true}.orElse({false})`.

And, I don't get how this would make the test execution re-run. Aren't we missing something like `upToDateWhen { !rerun.get() }`?

Reply:
> Also, .orElse(false) will eagerly observe the system property at configuration time defeating instant execution cache.
It should be .orElse(project.providers.provider { false }).

uh. This should really be simplified with a higher level API. We can't expect users to understand this.
`project.objects.systemProperty("idea.active", false")` could do the right thing in a concise, understandable DSL.

Reply:
Better now?

Reply:
@melix your suggestion sounds good, even more convenient than `provider.asPresence()` in that case

/cc @bamboo